fenix/app/src/main/java/org/mozilla/fenix/browser/StandardSnackbarErrorBinding.kt

```diff
@@ -0,0 +1,97 @@
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import mozilla.components.lib.state.helpers.AbstractBinding
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.FenixSnackbar
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.components.appstate.AppState
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.getRootView
+
+/**
+ * A binding that shows standard snackbar error.
+ */
+@OptIn(ExperimentalCoroutinesApi::class)
+class StandardSnackbarErrorBinding(
+    private val activity: Activity,
+    appStore: AppStore,
+) : AbstractBinding<AppState>(appStore) {
+
+    override suspend fun onState(flow: Flow<AppState>) {
+        flow.map { state -> state.standardSnackbarError }
+            .distinctUntilChanged()
+            .collect { standardSnackbarError ->
+                if (standardSnackbarError?.standardSnackbarErrorVisibility == true) {
+                    activity.getRootView()?.let { view ->
+                        val snackBar = FenixSnackbar.make(
+                            view = view,
+                            duration = Snackbar.LENGTH_INDEFINITE,
+                            isDisplayedWithBrowserToolbar = true,
+                        )
+                        snackBar.setText(
+                            standardSnackbarError.standardSnackbarErrorMessage
+                                ?: "",
+                        )
+                        snackBar.setButtonTextColor(
+                            ContextCompat.getColor(
+                                activity,
+                                R.color.fx_mobile_text_color_primary,
+                            ),
+                        )
+                        snackBar.setBackground(
+                            AppCompatResources.getDrawable(
+                                activity,
+                                R.drawable.save_to_pdf_error_snackbar_background,
+                            ),
+                        )
+                        snackBar.setSnackBarTextColor(
+                            ContextCompat.getColor(
+                                activity,
+                                R.color.fx_mobile_text_color_warning,
+                            ),
+                        )
+                        snackBar.setAction(
+                            text = activity.getString(R.string.standard_snackbar_error_dismiss),
+                            action = {
+                                view.context.components.appStore.dispatch(
+                                    AppAction.ShowStandardSnackbarErrorAction(
+                                        StandardSnackbarError(
+                                            false,
+                                            activity.getString(
+                                                R.string.unable_to_save_to_pdf_error,
```

Comment:
The snackbar text should be "Unable to generate PDF" in this case.

Reply:
Thanks for double checking!

Reply:
If the idea is to remove the snackbar should we dispatch  `dispatch(AppAction.ShowStandardSnackbarErrorAction(null))`? As just want to clear up the state.  And instead of calling  `ShowStandardSnackbarErrorAction` we could have `UpdateStandardSnackbarErrorAction`?